lib/events.js

```diff
@@ -332,6 +332,7 @@ function _onceWrap(target, type, listener) {
   var state = { fired: false, wrapFn: undefined, target, type, listener };
   var wrapped = onceWrapper.bind(state);
   wrapped.listener = listener;
+  listener.oneshot = true;
```

Comment:
I'm not keen on tacking this onto the user supplied function. I'd rather see a less invasive solution. Perhaps pass a new 'isOnce' boolean argument to `'newListener'` event handlers?